sklearn/metrics/_classification.py

```diff
@@ -3601,9 +3612,14 @@ def brier_score_loss(
         transformed_labels, y_proba = _validate_multiclass_probabilistic_prediction(
             y_true, y_proba, sample_weight, labels
         )
-
-    brier_score = np.average(
-        np.sum((transformed_labels - y_proba) ** 2, axis=1), weights=sample_weight
+        transformed_labels = xp.asarray(transformed_labels, device=device)
+        y_proba = xp.asarray(y_proba, device=device)
```

Comment:
Shouldn't these be on the device already. I think y_proba might be shifted to cpu because of the check_array function but assuming that y_true and y_prob are on the expected device transformed_labels should be on the device as well. Or is this just handling for the array-api-strict?

Reply:
> y_proba might be shifted to cpu because of the check_array 

In which case could `y_proba` be shifted to cpu?

If it is possible that `check_array` alters the device, we probably need to do `get_namespace_and_device` before any `check_array`s - i.e., in `_validate_binary_probabilistic_prediction` we do a `column_or_1d` first, which does `check_array` inside. I wonder if it would be good practice to just always do `get_namespace_and_device` first?

> Or is this just handling for the array-api-strict?

Question, why would it be needed for `array-api-strict`?